### PR TITLE
feat: Add gem target

### PR DESCRIPTION
**Pull request by HazAT**
I successfully did released https://github.com/getsentry/raven-ruby/releases/tag/3.0.2 with this branch.

```
➜ node ~/Projects/craft/dist/index.js publish 3.0.2
ℹ info craft 0.10.1
ℹ info "craft" version is compatible with the minimal version from the configuration file.
ℹ info Publishing version: "3.0.2"
ℹ info Using "GithubStatusProvider" for status checks
ℹ info Using "ZeusArtifactProvider" for artifacts
ℹ info No config provided for Github status provider, calculating the combined status...
ℹ info Revision df427b7320d36f30975d6746f730b1d12316bbf6 has been built successfully.
ℹ info
ℹ info Available artifacts:
┌────────────────────────┬─────────┬──────────────────────────────────┬─────────────────────┐
│ File Name              │ Size    │ Updated                          │ ContentType         │
├────────────────────────┼─────────┼──────────────────────────────────┼─────────────────────┤
│ sentry-raven-3.0.2.gem │ 46.5 kB │ 2020-08-20T10:11:00.038860+00:00 │ application/tar+gem │
└────────────────────────┴─────────┴──────────────────────────────────┴─────────────────────┘

ℹ info Publishing to targets:
ℹ info   - gem
ℹ info   - github
ℹ info   - registry
ℹ info
? Is everything OK? Type "yes" to proceed: yes
ℹ info
ℹ info =================================
ℹ info === Publishing to target: gem ===
ℹ info =================================
ℹ info [gem] › Pushing gem "sentry-raven-3.0.2.gem"
ℹ info [gem] › Successfully registered gem
ℹ info
ℹ info ====================================
ℹ info === Publishing to target: github ===
ℹ info ====================================
ℹ info [github] › Target "github": publishing version "3.0.2"...
ℹ info [github] › Git tag: "3.0.2"
ℹ info [github] › Creating a new release for tag "3.0.2"
ℹ info [github] › Uploading asset "sentry-raven-3.0.2.gem" to getsentry/raven-ruby:3.0.2
❯ log [github] › Uploaded asset "sentry-raven-3.0.2.gem".
ℹ info
ℹ info ======================================
ℹ info === Publishing to target: registry ===
ℹ info ======================================
ℹ info [registry] › Cloning "https://github.com/getsentry/sentry-release-registry/" to "/var/folders/7d/4fxc2_s94_z6ygyf5yk74t700000gn/T/craft-release-registry-v1HQuZ"...
ℹ info [registry] › Adding the version file to the registry for canonical name "gem:sentry-raven"...
ℹ info [registry] › Adding extra data (checksums, download links) for available artifacts...
ℹ info [registry] › Pushing the changes...
ℹ info [registry] › Release registry updated
ℹ info
ℹ info Merging release branch: "release/3.0.2" into "master"...
ℹ info Merging: done.
ℹ info Removed the remote branch: "release/3.0.2"
✔ success Version 3.0.2 has been published!
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -822,6 +823,30 @@ statusProvider:
       - Travis CI - Branch # or whatever builds and pushes your source image
 ```
 
+### Ruby Gems Index (`gem`)
+
+Pushes a gem [Ruby Gems](https://rubygems.org).
+It also requires you to be logged in `gem login`.
```
Comment (tonyo) [inline README.md:829]:
```suggestion
It also requires you to be logged in with `gem login`.
```

Comment (tonyo) [reply README.md:829]: For the future, Is there an option to provide a RubyGems API token to `gem` command, so we don't have to go through `gem login` process?

File: README.md [hunk 2]
```diff
@@ -822,6 +823,30 @@ statusProvider:
...
+
+**Environment**
+
+`gem` must be installed on the system.
+
+| Name      | Description                                  |
+| --------- | -------------------------------------------- |
+| `GEM_BIN` | **optional**. Path to gem. Defaults to `gem` |
```
Comment (tonyo) [inline README.md:837]:
```suggestion
| `GEM_BIN` | **optional**. Path to "gem" executable. Defaults to `gem` |
```

File: src/targets/gem.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import { BaseTarget } from './base';
+
+const logger = loggerRaw.withScope('[gem]');
+
+const DEFAULT_GEM_BIN = 'gem';
+
+/**
+ * Command to launch gem
+ */
+const GEM_BIN = process.env.TWINE_BIN || DEFAULT_GEM_BIN;
```
Comment (tonyo) [inline src/targets/gem.ts:17]:
```suggestion
const GEM_BIN = process.env.GEM_BIN || DEFAULT_GEM_BIN;
```

**Review by tonyo — CHANGES_REQUESTED** [review]
Nice! 
Minor suggestions.
And CHANGELOG, please.

File: README.md (hunk 1 above)
Comment (HazAT) [reply README.md:829]: I think you can export an env variable but we need to check that.

**Review by tonyo — APPROVED** [review]

**Pull request merged**